### PR TITLE
Make it possible to delete allocations from front-end

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR emulates the shift page so that Allocations can be removed.

### CODE REVIEW

File: static/js/components/AllocationPage.jsx [hunk 1]
```diff
@@ -21,6 +25,23 @@ const useStyles = makeStyles((theme) => ({
...
+  },
+  allocationDeleteDisabled: {
+    cursor: "pointer",
+    fontSize: "2em",
+    position: "absolute",
+    padding: 0,
+    right: 0,
+    top: 0,
+    opacity: 0,
+  },
```
Comment (guynir42) [inline static/js/components/AllocationPage.jsx:44]: I'm not sure, but I think when the button is disabled, you should get the properties of both these classes, so you really only need the opacity to be different (please check if that works!).

**Review by guynir42 — APPROVED** [review]
Very minor comment.

File: static/js/components/AllocationPage.jsx (hunk 1 above)
Comment (mcoughlin) [reply static/js/components/AllocationPage.jsx:44]: I dont understand this comment :)

Comment (guynir42) [reply static/js/components/AllocationPage.jsx:44]: I'm not sure: but the way the button is defined, I think it gets the `allocationDelete` class applied as root, and when it is disabled it applies `allocationDeleteDisabled` class on top of the root. If that's true then you don't need to have all the properties copied over from one class to the other, only the opacity which is different. I am not sure if this is how those definitions get inherited but it's worth a try.

Comment (mcoughlin) [reply static/js/components/AllocationPage.jsx]: This was very correct. didn't need any of it!

**Comment by mcoughlin** [comment]
Very minor (addressed) comments. Merging.

**Pull request merged**